src/main/java/dog/pawbook/ui/EntityCard.java

```diff
@@ -38,13 +45,31 @@
     private Label id;
     @FXML
     private FlowPane tags;
+    @FXML
+    private Label staticId;
+    @FXML
+    private ImageView entityImage;
+
+
 
     /**
      * Creates a {@code EntityCard} with the given {@code Entity} and index to display.
      */
     public EntityCard(Entity entity, int displayedId) {
         super(FXML);
         this.entity = entity;
+
+        this.entityImage.setFitHeight(50);
+        this.entityImage.setFitWidth(50);
+
+        if (entity instanceof Dog) {
+            this.entityImage.setImage(dogImage);
+        } else if (entity instanceof Owner) {
+            this.entityImage.setImage(ownerImage);
+        } else {
+            this.entityImage.setImage(programImage);
+        }
```

Comment:
is it better if program is explicitly matched as well? then you get to do some defensive coding by asserting in the else

Reply:
Done.